### PR TITLE
FIX ignore hidden files when loading datasets, solvers, etc

### PR DESCRIPTION
**Pull request by antoinecollas**
When working on remote machines (e.g. with sshfs), macos automatically creates hidden files.
For example, if there is a file `datasets/mnist.py`, then it automatically creates a metadata file `datasets/._mnist.py`.
Thus, benchopt loads `datasets/mnist.py` but also `datasets/._mnist.py` which creates errors since  `datasets/._mnist.py` only contains metadata. To prevent this, this PR changes the regex that loads submodules to ignore hidden files.

### CODE REVIEW

**Review by agramfort — APPROVED** [review]

**Comment by mathurinm** [comment]
Should we add a UT ?

**Comment by tomMoral** [comment]
Yes, can you add a test? You can use something similar to what is done here: https://github.com/benchopt/benchopt/blob/79e4608b6a26a02d832668bc7d18b1960b311b69/benchopt/tests/test_benchmark_features.py#L54C1-L54C1

File: benchopt/tests/test_benchmark_features.py [hunk 1]
```diff
@@ -99,3 +100,32 @@ def raise_error():
             )
     finally:
         os.environ['BENCHOPT_RAISE_INSTALL_ERROR'] = prev_value
+
+
+def test_ignore_hidden_files():
```
Comment (tomMoral) [inline benchopt/tests/test_benchmark_features.py:105]: I just simplified a bit this test but it looks super good! thanks @antoinecollas

**Review by tomMoral — APPROVED** [review]
LGTM!

**Pull request merged**